traffic_ops/traffic_ops_golang/staticdnsentry/staticdnsentry.go

```diff
@@ -81,8 +82,8 @@ func (staticDNSEntry *TOStaticDNSEntry) SetKeys(keys map[string]interface{}) {
 // Validate fulfills the api.Validator interface
 func (staticDNSEntry TOStaticDNSEntry) Validate() error {
 	errs := validation.Errors{
-		"host":    validation.Validate(staticDNSEntry.Host, validation.Required),
-		"address": validation.Validate(staticDNSEntry.Address, validation.Required),
+		"host":    validation.Validate(staticDNSEntry.Host, validation.Required, is.DNSName),
+		"address": validation.Validate(staticDNSEntry.Address, validation.Required, is.Host),
 		"dsname":  validation.Validate(staticDNSEntry.DeliveryService, validation.Required),
```

Comment:
Why is the dsname (which maps the to xml_id) required? Shouldn't it be the DS ID since that's what's used in the insert statement?

Same goes for "type", which is the type name -- I think "typeId" should be required instead.